controller/src/beerocks/master/tasks/optimal_path_task.cpp

```diff
@@ -227,8 +227,88 @@ void optimal_path_task::work()
+                              << " is not on the candidate ap list, continue as usual.";
+        }
+
+        if (database.get_client_stay_on_selected_band(client) == eTriStateBool::ENABLE) {
+            TASK_LOG(INFO) << "Client stay on selected bands enabled";
+            if (!database.is_hostap_on_client_selected_bands(
+                    client, tlvf::mac_from_string(current_hostap))) {
+                TASK_LOG(INFO) << "Current radio " << current_hostap
+                               << " is not on one of client's selected bands "
+                               << int(database.get_client_selected_bands(client));
+                // Try to find radio with selected bands first on local device (same device the client
+                // is currently connected on) and force steer the client to that radio.
+                auto current_hostap_siblings = database.get_node_siblings(current_hostap);
+                auto sibling_it =
+                    std::find_if(current_hostap_siblings.begin(), current_hostap_siblings.end(),
+                                 [&](const std::string &sibling) {
+                                     return database.is_hostap_on_client_selected_bands(
+                                         client, tlvf::mac_from_string(sibling));
+                                 });
+
+                if (sibling_it != current_hostap_siblings.end()) {
+                    chosen_bssid =
+                        database.get_hostap_vap_with_ssid(sibling_it->data(), current_hostap_ssid);
+                    state          = SEND_STEER_ACTION;
+                    is_force_steer = true;
+                    TASK_LOG(INFO) << "Found local radio " << sibling_it->data()
+                                   << " on selected bands, force steer client to that radio";
+                    break;
+                }
+                TASK_LOG(WARNING) << "Couldnt find local radio on selected bands "
+                                  << int(database.get_client_selected_bands(client))
+                                  << " with same client's ssid " << current_hostap_ssid;
+            }
+            // In case client is already connected to one of the selected bands
+            // continue with optimal path task but remove all non selected band hostaps
+            // from steering candidate list.
+            remove_all_client_non_selected_band_radios(potential_11k_aps, client);
+        }
+
+        // hostap's list is ready , lets check if we have candidates left
+        const auto hostap_candidates_size = potential_11k_aps.size();
+        if (hostap_candidates_size == 0) {
+            TASK_LOG(WARNING) << "Candidates list is empty, aborting optimal path task";
+            finish();
+            return;
+        }
+
+        if ((hostap_candidates_size == 1) && (potential_11k_aps.begin()->first == current_hostap)) {
+            TASK_LOG(WARNING)
```

Comment:
```suggestion
            TASK_LOG(DEBUG)
```

Reply:
why?

Reply:
never mind , changed to debug

Reply:
since WARNING should indicate that this situation should not be reached, but we can waver and not fail on it.
in this case, I think the scenario is a feasible scenario - if you think it doesn't leave it as WARNING.